src/v/http_test_utils/registered_urls.cc

```diff
@@ -0,0 +1,115 @@
+}
+
+void http_test_utils::registered_urls::add_mapping::add_mapping_when::
+  then_reply_with(ss::httpd::reply::status_type status) {
+    then_reply_with(default_response, status);
+}
+
+void http_test_utils::registered_urls::add_mapping::add_mapping_when::
+  then_reply_with(ss::sstring content, ss::httpd::reply::status_type status) {
+    if (!r.contains(url)) {
+        r[url] = method_reply_map{};
+    }
+
+    if (!r[url].contains(method)) {
+        r[url][method] = content_reply_map{};
+    }
+
+    r[url][method][request_content] = response{
+      .body = std::move(content), .status = status};
+}
+
+http_test_utils::registered_urls::add_mapping::add_mapping_when&
+http_test_utils::registered_urls::add_mapping::add_mapping_when::with_method(
+  ss::operation_type m) {
+    method = m;
+    return *this;
+}
+
+http_test_utils::registered_urls::add_mapping::add_mapping_when&
+http_test_utils::registered_urls::add_mapping::add_mapping_when::with_content(
+  ss::sstring content) {
+    request_content = std::move(content);
+    return *this;
+}
+
+http_test_utils::registered_urls::add_mapping::add_mapping_when
+http_test_utils::registered_urls::request(
+  ss::sstring url,
+  ss::httpd::operation_type method,
+  ss::sstring request_content) {
+    return add_mapping::add_mapping_when{
+      .r = request_response_map,
+      .url = std::move(url),
+      .method = method,
+      .request_content = std::move(request_content)};
+}
+
+http_test_utils::registered_urls::add_mapping::add_mapping_when
+http_test_utils::registered_urls::request(
+  ss::sstring url, ss::httpd::operation_type method) {
+    return request(std::move(url), method, default_content);
+}
+
+http_test_utils::registered_urls::add_mapping::add_mapping_when
+http_test_utils::registered_urls::request(ss::sstring url) {
+    return request(std::move(url), ss::httpd::GET, default_content);
+}
+
+http_test_utils::response
+http_test_utils::registered_urls::lookup(ss::httpd::const_req& req) const {
+    auto url = req._url;
+    auto it = request_response_map.find(url);
+    if (it == request_response_map.end()) {
+        return not_found;
+    }
+
+    auto method_mapping = it->second;
+    auto m_it = method_mapping.find(ss::httpd::str2type(req._method));
+    if (m_it == method_mapping.end()) {
+        return not_found;
+    }
+
+    auto content_mapping = m_it->second;
+    auto content = req.content;
+    if (auto c_it = content_mapping.find(content);
+        c_it != content_mapping.end()) {
+        return c_it->second;
+    }
+
+    vassert(
+      content_mapping.contains(default_content),
+      "The request map is not set with a default content key: {}",
+      default_content);
+    return content_mapping[default_content];
+}
+
+std::ostream&
+operator<<(std::ostream& os, const http_test_utils::response& resp) {
+    fmt::print(
+      os, "{status: {}, body: {}}", static_cast<uint>(resp.status), resp.body);
```

Comment:
`body: {}}"` do you need a double `}}` at the end for a literal `}` I can't remember if this is a special case or not.

Reply:
yes it requires double braces (I faced the same issue on the IAM roles PR when I used one brace), fixed